src/odemis/driver/orsay.py

```diff
@@ -2769,3 +2781,254 @@ def terminate(self):
             self.stop()
             self._executor.shutdown()
             self._executor = None
+
+
+# TODO These values should be implemented in the Orsay API under HybridAperture.XAxis/YAxis.Position.Min/Max when
+#  these are implemented use those values instead.
+APERTURE_AXIS_RANGE = (-6e-3, 6e-3)
+
+
+class FIBAperture(model.Actuator):
+    """
+    Represents the Aperture carrier of the Orsay FIB. This class allows controls of the current position of the
+    aperture carrier using the move functions like a stage. This allows positioning of a certain aperture w.r.t. the
+    ion beam. The aperture carrier is a plate which contains multiple apertures.
+    Using the selectedAperture VA an aperture can be selected using the number it is identified with. The size of the
+    currently selected aperture can be found using the sizeSelectedAperture VA.
+    """
+    def __init__(self, name, role, parent, **kwargs):
+        axes_def = {"x": model.Axis(unit="m", range=APERTURE_AXIS_RANGE),
+                    "y": model.Axis(unit="m", range=APERTURE_AXIS_RANGE)}
+        super().__init__(name, role, parent=parent, axes=axes_def, **kwargs)
+        self._hybridAperture = self.parent.datamodel.HybridAperture
+        self._nmbr_apertures = int(self._hybridAperture.SelectedDiaph.Max)
+
+        self._selectedApertureConnector = None
+        self.selectedAperture = model.IntContinuous(int(self._hybridAperture.SelectedDiaph.Actual),
+                                                    readonly=True, range=(0, self._nmbr_apertures))
+        self._sizeSelectedApertureConnector = None
+        self.sizeSelectedAperture = model.FloatVA(float(self._hybridAperture.SizeOfSelectedDiaph.Actual),
+                                                  readonly=True)
+
+        self._apertureDict = {}
+        self._apertureConnectors = {}
+        # Dict to convert the strings returned by the Orsay param to the real type.
+        self._apertureVarTypes = {"LifeTime": int, "Size": float, "PositionX": float, "PositionY": float}
+        for aprtr_nmbr in range(self._nmbr_apertures):
+            self._apertureDict[aprtr_nmbr] = {
+                "LifeTime": int(0),
+                "Size": float(0.0),
+                "Position": {"x": float(0.0), "y": float(0.0)}
+            }
```

Comment:
That doesn't work because that function doesn't create the structure of the dict itself. It assumes the structure is already present. 
For clarity and readability I separated both so that if you look how the aperture dict is defined you are redirected to the init with a initialization of the structure and types of the dict.
What do you think I should do with it?